### ISSUE TITLE
NoAVX Machine-learning Version

### ISSUE THREAD

**Issue by Synoxion** [imagegenius/docker-immich#30]
Is it possible to add a version with bertmelis Immich machine learning for x86 without AVX? #[bertmelis]/[immich-machine-learning-no-avx]

**Comment by hydazz** [comment]
That project is outdated and immich now uses torch, I believe to compile touch without AVX you need a CPU without AVX (I may be wrong). I will try this, would you be willing to test it/compile torch for us. I assume you have a system that does not have AVX?

**Comment by Synoxion** [comment]
Yes, that is unfortunately correct. All Intel CPUs up to the Pentium Gold G6400 / Celeron G5900 do not have AVX. I still use the G4560, which is still almost perfect for Unraid. This one has ECC. 
But compiling should also work on AVX CPUs with the appropriate flags. This is what I found for pytorch on the net. Here someone has built for an AMD CPU.

-D CMAKE_BUILD_TYPE=Release \
      -D USE_AVX=OFF \
      -D USE_NNPACK=OFF \
      -D USE_MKLDNN=OFF \
      -D C_HAS_AVX_2=OFF \
      -D C_HAS_AVX2_2=OFF \
      -D CXX_HAS_AVX_2=OFF \
      -D CXX_HAS_AVX2_2=OFF \
      -D CAFFE2_COMPILER_SUPPORTS_AVX512_EXTENSIONS=OFF \

But I have no idea what the options -D USE_NNPACK=OFF and -D USE_MKLDNN=OFF do.
I'm not that knowledgeable in compiling, unfortunately. I have only tried it twice. With success, but it was a long time ago.
If again expect nevertheless a CPU without AVX is necessary, I could try me at it.
